src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js

```diff
@@ -28,17 +32,25 @@ export const useArrowNav = (selected, setSelected) => {
   }, [selected, setSelected]);
 
   const detectKeyDown = (e) => {
+    const problemTypeValues = Object.values(ProblemTypeKeys);
+    const advanceTypeValues = Object.values(AdvanceProblemKeys);
     switch (e.key) {
       case 'ArrowUp':
-        if (ProblemTypes[selected].prev) {
+        if (problemTypeValues.includes(selected) && ProblemTypes[selected].prev) {
           setSelected(ProblemTypes[selected].prev);
           document.getElementById(ProblemTypes[selected].prev).focus();
+        } else if (advanceTypeValues.includes(selected) && AdvanceProblems[selected].prev) {
+          setSelected(AdvanceProblems[selected].prev);
+          document.getElementById(AdvanceProblems[selected].prev).focus();
```

Comment:
I just realized that Form.Radio has the arrow functionality coded in... I'm so sorry I did not realize that earlier. You could remove the `else if` with advanced types and arrow navigation will still work.

The type checking looks great.